dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,395 @@
+
+Since the format of the transaction remained the same only the optional data is
+described in the following tables. Note that there are two valid formats based
+on the number of treasury spend votes that are occurring simultaneously.
+
+Format 1
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[0]<code>OP_RETURN</code><br>[1]<code>OP_DATA_XX</code><br>[2..3]<code>'TV'</code><br>[4..]<code>treasury vote...</code>||Up to two (2) treasury votes can be encoded using this format.
+|}
+
+Format 2
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[0]<code>OP_RETURN</code><br>[1]<code>OP_PUSHDATA1</code><br>[2]data length<br>[5...]<code>treasury vote...</code>||Up to seven (7) treasury votes can be encoded.
+|}
+
+Treasury vote
+{|
+!Transaction!!Script!!Description
+|-
+|TxOut[N-1]||[M] TSpend transaction hash<br>[M+32] byte vote bits||Yes vote = 0x01<br>No vote = 0x02<br>All other values shall be rejected rendering the overall transaction invalid.
+|}
+
+
+===Debit treasury===
+
+The TSpend transactions are rather complex and some additional definitions need
+to be made prior to explaining the format.
+
+Required chain configuration parameters:
+* Treasury Vote Interval, or TVI, determines in which blocks a TSpend may occur. A block is considered on a TVI when the block height modulo TVI is equal to zero.
+* Treasury Vote Interval Multiplier, or TVIM, dictates the total duration (total duration = TVI * TVIM) where votes for or against the expenditure are considered valid.
+* Treasury Vote Quorum Multiplier and Divisor, TVQM, and TVQD respectively, are used to calculate the required number of votes to reach quorum.
+* Treasury Vote Required Multiplier and Divisor, TVRM and TVRD respectively, are used to calculate the required number of Yes votes.
+* Treasury Expenditure Policy, or TEP, determines the overall duration where all treasury expenditures are summed and averaged.
+* Treasury Expenditure Window, or TEW, is used to calculate the window that is used to verify that the sum of all TSpends does not exceed ~150% of the average treasury spending.
+* Treasury Expenditure Bootstrap, or TEB, is used to determine a baseline expenditure in order to bootstrap treasury payouts.
+* Politeia Keys, or PiKeys, are a number of well-known secp256k1 public keys that are allowed to sign TSpend transactions.
+
+Only the Pi key administrator can author transactions that debit the treasury
+account and generate UTXOs for recipients.
+
+The life-cycle of a TSpend transaction is as follows
+# PiKey administrator creates a TSpend transaction with an Expiry in the future.
+# The TSpend is signed and broadcast to the network.
+# If the TSpend is nominally valid (Expiry in the future, valid signature, valid public key etc) it is allowed in the mempool.
+# Once the TSpend hits the mempool wallets will be notified.
+# Voting wallets with a voting policy will now start voting either yes/no or abstain at the TSpend vote window start.
```

Comment:
I would remove "now" as it tricked me to think wallets will start voting the moment they see TSpend in the mempool.

Reply:
Fixed